docs/architecture/transactions/kernel.md

```diff
@@ -1,54 +1,95 @@
+### Account data
 
-Fifth, the **input notes** are being processed. This involves per note reading the data from the advice provider and storing it at the appropriate memory addresses. Next to the total number of consumed notes, input note data consists of its serial number, the roots of the script, inputs and asset vault, its metadata and all its assets. As each note is consumed its hash and nullifier is computed. The transaction Nullifier commitment is computed via a sequential hash of all `(nullifier, ZERO)` pairs for all consumed notes. This step involves authentication that the input note data provided via the advice provider is consistent with the chain history.
+Account data processing involves reading the data from the advice provider and storing it at the appropriate memory addresses. The account data consists of account vault roots, storage, and code. 
 
-_Note: One needs to provide the note data to compute the Nullifier, e.g. the [note script](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#script) and the [serial number](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#serial-number). So one needs to know the note data to execute the prologue of a transaction. This is how the [note recipient](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#note-recipient) defines the set of users who can consume a specific note. The executing account needs to provide the pre-image data to the recipient at the time of execution._
+As the account data is read from the advice provider, the account hash is computed. If the account is new then the global initial account hash is updated and the new account is validated. If the account already exists then the computed account hash is asserted against the account hash provided via global inputs. It is also asserted that the account id matches the account id provided via the global inputs (`operand_stack`).
 
-Lastly, if a transaction script is provided, its root is being stored at the pre-defined memory address.
+### Input note data
 
-## The Note Processing
-If there are input notes they are being consumed in a loop. For every note, the [MAST root](https://0xpolygonmiden.github.io/miden-vm/design/programs.html) of the note script is being loaded onto the stack. Then, by calling a [`dyncall`](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/code_organization.html?highlight=dyncall#dynamic-procedure-invocation) the note script is being executed in a new context to prevent unwanted memory access.
+Input note processing involves each note reading the data from the advice provider and storing it at the appropriate memory addresses. 
```

Comment:
This reads to me as if the note would be reading data

```
... each note reading the data from the advice provider ...
```

It is more like the kernel reads the data from each note (which is provided via the advice provider). Basically, the kernel reads all data from the advice provider (and global inputs) and stores it into memory. All data means, all note data, and account data, and blockchain data.